### DISCUSSION TITLE
Show folders in file picker

### DISCUSSION THREAD

**Discussion by LasseRosenow** [directus/directus#10940] [Archive]
I have my media gallery very well organized.

![image](https://user-images.githubusercontent.com/10547444/148700820-51e999a3-d13c-4378-9231-0bd23fd7a36d.png)

As you can see, all my files are in folders except 4.

But when I want to select a file from the root library this happens:

![image](https://user-images.githubusercontent.com/10547444/148700864-823b125a-dfaa-44e7-b388-b17beea6eb4b.png)

I understand, that it makes sense that I am able to select all files that are inside any sub directory, but I don't understand, why this list needs to be flattened.

My proposal would be to show folders instead of all the images, so that it is not as much of a struggle anymore

**Comment by benhaynes** [comment]
I agree, and I think there's already a discussion in place for this. Folder selection needs to be included in all media drawers.

**Reply by LasseRosenow** [reply]
Uh that sounds nice, I was searching for it but I couldn't find it. Sorry if its a duplicate then

**Comment by LasseRosenow** [comment]
Actually I have a small question.

As I found the original discussion about this topic, I saw this image that you posted:

Source: https://github.com/directus/directus/discussions/3464

![image](https://user-images.githubusercontent.com/10547444/148709184-d0844704-a77c-4c24-9dee-d16b5bd45d39.png)

I think you just posted to visualise what you meant with showing all folders at the top. Though the image itself is about showing some groups of collection items at the top to filter some kind of collection.

Now I did not know this was possible and I just can't figure out how to do it, so I was wondering if this is a thing or if it is planned or something? I could not find a discussion regarding filtering collections by groups either 😅

**Reply by benhaynes** [reply]
Just planned.  I would like to include an option for showing folders and other relational groups within content layouts. But I also think we should include the tree view sidebar for files in drawers. So I guess we would have both.

**Reply by LasseRosenow** [reply]
That sounds truly amazing, can't wait to see this happening :) Thanks so much :)

**Comment by polmonroig** [comment]
Is there any update on this?? Is it a task in progress? Is it already released in some version? I wanted to migrate from Strapi to Directus since it is much better in all aspects, but this specific folder scenario is the only thing that might make me think twice about it since I have a lot of media files, and have to pick them up all mixed up is complicated and the left sidebar is not so intuitive to navigate.

Thanks

**Reply by benhaynes** [reply]
No updates, but anything can be sponsored to get it across the finish line based on the timelines of external users/companies. 👍

**Comment by notflip** [comment]
Is there any update on this?

**Comment by w0kyj** [comment]
Implemented by https://github.com/directus/directus/pull/18320
